docs/docs/getting_started/tooling/noir_codegen.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Noir Codegen for Typescript
+description: Learn how to use Noir codegen to generate Typescript bindings 
+keywords: [Nargo, Noir, compile, TypeScript]
+sidebar_position: 2
+---
+
+You can generate TypeScript bindings with `noir_codegen`, allowing for easier and faster integration of your Noir circuits into TypeScript projects.
+
+It is used in conjunction with `nargo export` which generates JSON for specified functions in a Noir library (not binary or contract circuits).
```

Comment:
```suggestion
There are two steps to this process. First, you will run `nargo export`, which compiles the JSON for decorated functions. And then, you will install and use `noir_codegen` which outputs the TS file.

:::warning

This process only works for Nargo _libraries_, i.e. the ones with `type = "lib"` in their `Nargo.toml` file. Binaries or contracts are not supported.

:::
```

Reply:
Similarly I'll clarify this a little more.